flaml/automl.py

```diff
@@ -108,6 +108,7 @@ def valid_starting_point(self, starting_point, search_space):
                     value, search_space[name].get("domain")
                 )
                 for name, value in starting_point.items()
+                if name != "FLAML_sample_size"
```

Comment:
It's not sufficient to exclude "FLAML_sample_size" from the starting_point. The specified sample size needs to be used as the starting sample size.